packages/typespec-azure-resource-manager/test/enforce-constraint.test.ts

```diff
@@ -0,0 +1,77 @@
+import { expectDiagnosticEmpty, expectDiagnostics } from "@typespec/compiler/testing";
+import { describe, it } from "vitest";
+import { checkFor } from "./test-host.js";
+
+describe("typespec-azure-resource-manager: @enforceConstraint", () => {
+  it("emits no error when template param extends from Resource", async () => {
```

Comment:
I think we need some additional tests
- direct extension of Resource (succeed)
- model the *is* Resource (fail)